bin/sign-images.py

```diff
@@ -0,0 +1,175 @@
+        with open(file_path, "wb") as key_file:
+            key_file.write(decoded_key)
+        os.chmod(file_path, 0o600)  # Set secure permissions
+        return True
+    except (base64.binascii.Error, OSError) as e:
+        print(f"Error decoding or writing key from {env_var_name}: {e}")
+        sys.exit(1)
+
+
+def docker_image_exists(repository, tag):
+    """Check if an image exists in Docker Hub."""
+    if "quay.io/astronomer/" in repository:
+        # Convert quay.io repository to Docker Hub format
+        dockerhub_repo = repository.replace("quay.io/astronomer/", "astronomerinc/")
+        # Check if the image exists on Docker Hub
+        try:
+            # Use Docker Hub API to check if the image exists
+            dockerhub_api_url = f"https://hub.docker.com/v2/repositories/{dockerhub_repo}/tags/{tag}"
+            response = requests.head(dockerhub_api_url, timeout=100)
+            return response.status_code == 200
+        except requests.RequestException:
+            return False
+    return False
+
+
+def sign_image(repo, tag, sha, key_path, password=None):
+    """Sign a container image with cosign."""
+    full_image = f"{repo}:{tag}"
+    digest_reference = f"{full_image}@sha256:{sha}"
+
+    print(f"Signing image: {full_image} with SHA: {sha}")
+    env = os.environ.copy()
+    if password:
+        env["COSIGN_PASSWORD"] = password
+    repositories_to_sign = [repo]
+    # Checking if the Docker Hub version exists
+    if "quay.io/astronomer/" in repo and docker_image_exists(repo, tag):
+        # Replace quay.io with Docker Hub repository
+        dockerhub_repo = repo.replace("quay.io/astronomer/", "astronomerinc/")
+        repositories_to_sign.append(dockerhub_repo)
+        print(f"Found matching Docker Hub image: {dockerhub_repo}:{tag}")
+    for repository in repositories_to_sign:
+        full_image = f"{repository}:{tag}"
+        digest_reference = f"{full_image}@sha256:{sha}"
+        try:
+            subprocess.run(
+                ["cosign", "verify", "--insecure-ignore-tlog", "--key", f"{key_path}.pub", digest_reference],
+                check=True,
+                capture_output=True,
+            )
+            print(f"Image already signed: {full_image}")
+            continue  # Skip if already signed
+        except subprocess.CalledProcessError:
+            pass  # Image is not yet signed
+        sign_cmd = ["cosign", "sign", "--key", key_path, digest_reference]
+        try:
+            subprocess.run(sign_cmd, env=env, check=True)
+            print(f"✓ Signed {full_image}")
+        except subprocess.CalledProcessError as e:
+            print(f"Error signing {full_image}: {e}")
+            # Continue with other images instead of exiting
+            continue
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Sign container images in an Astronomer release.")
+    parser.add_argument("--version", "-v", help="Version tag to sign (e.g., 0.36.0)", required=False)
```

Comment:
Can we change this to `--tag`? `--version` is usually used to check the version of the CLI tool that is being run.

Reply:
Oh, wait, `--tag` doesn't make sense in the context of the Astronomer "Software" version. I recently implemented this in `software-vuln-exporter` as `--release`, but I honestly don't know if that is any better than `--version` https://github.com/astronomer/software-vuln-exporter/blob/main/src/software_vuln_exporter/cli.py#L20-L24

We can leave this as `--version` if we want. I don't think there's a clear better option.